skmultiflow/classification/meta/adaptive_random_forests.py

```diff
@@ -0,0 +1,396 @@
+        evaluator_method_metric: string (default='acc')
+            / Metric used to track tree performance.
+            / 'acc' - Accuracy
+        drift_detection_method: BaseDriftDetector default(ADWIN(0.001))
+            Drift Detection method
+        warning_detection_method: BaseDriftDetector default(ADWIN(0.01))
+            Warning Detection method
+        nominal_attributes: list, optional
+            List of Nominal attributes. If emtpy, then assume that all attributes are numerical.
+
+
+        Notes
+        -----
+        The 3 most important aspects of Adaptive Random Forest [1]_ are:
+        (1) inducing diversity through re-sampling;
+        (2) inducing diversity through randomly selecting subsets of features for node splits (see
+        skmultiflow.classification.trees.arf_hoeffding_tree);
+        (3) drift detectors per base tree, which cause selective resets in response to drifts.
+        It also allows training background trees, which start training if a warning is detected and replace the active
+        tree if the warning escalates to a drift.
+
+        References
+        ----------
+        .. [1] Heitor Murilo Gomes, Albert Bifet, Jesse Read, Jean Paul Barddal, Fabricio Enembreck,
+           Bernhard Pfharinger, Geoff Holmes, Talel Abdessalem.
+           Adaptive random forests for evolving data stream classification.
+           In Machine Learning, DOI: 10.1007/s10994-017-5642-8, Springer, 2017.
+
+    """
+
+    def __init__(self,
+                 nb_ensemble=10,
+                 feature_mode='sqrt',
+                 subspace_size=2,
+                 disable_background_learner=False,
+                 disable_drift_detection=False,
+                 disable_weighted_vote=False,
+                 lambda_value=6,
+                 evaluator_method_metric='acc',
+                 drift_detection_method: BaseDriftDetector=ADWIN(0.001),
+                 warning_detection_method: BaseDriftDetector=ADWIN(0.01),
+                 nominal_attributes=None):
+        """AdaptiveRandomForest class constructor."""
+        super().__init__()          
+        self.nb_ensemble = nb_ensemble        
+        self.feature_mode = feature_mode
+        self.subspace_size = subspace_size
+        self.disable_background_learner = disable_background_learner   
+        self.disable_drift_detection = disable_drift_detection        
+        self.disable_weighted_vote = disable_weighted_vote
+        self.lambda_value = lambda_value
+        self.drift_detection_method = drift_detection_method
+        self.warning_detection_method = warning_detection_method
+        self.instances_seen = 0
+        self._train_weight_seen_by_model = 0.0
+        self.ensemble = None
+        self.nominal_attributes = nominal_attributes
+        self._evaluator_method = ARFBaseClassifierEvaluator  # TODO use skmultiflow evaluator
+
+    def fit(self, X, y, classes=None, weight=None):
```

Comment:
I know it is not implemented in the other methods as well, but couldn't it call `partial_fit` in a loop for the given `X` and `y`?

Reply:
That is an interesting point to consider. Any comments @abifet  and @jmread ?